lib/pages/drive_detail/components/drive_detail_data_table_source.dart

```diff
@@ -0,0 +1,232 @@
+part of '../drive_detail_page.dart';
+
+class DriveDetailDataTableSource extends DataTableSource {
+  final List<DriveTableFolder> folders;
+  final List<DriveTableFile> files;
+  final BuildContext context;
+  DriveDetailDataTableSource(
+      {required this.folders, required this.files, required this.context});
+
+  @override
+  DataRow? getRow(int index) {
+    assert(index >= 0);
+
+    if (index >= (folders.length + files.length)) {
+      return null;
+    }
+
+    if (index < folders.length) {
+      final folder = folders[index];
+      return _buildFolderRow(
+        context: context,
+        folder: folder.folder,
+        onPressed: folder.onPressed,
+        index: index,
+        selected: folder.selected,
+      );
+    } else {
+      final file = files[folders.length > files.length
+          ? folders.length - index
+          : index - folders.length];
```

Comment:
Given that at this point `index >= folders.length`, is there any issue by doing the following?
```suggestion
      int fileIndex = index - folders.length;
      final file = files[fileIndex];
```